### PR TITLE
OKTA-520236 - Implementation for Resend Link that contains HTML

### PR DESCRIPTION
**Pull request by glenfannin-okta**
## Description:
Purpose of this PR is to fix the raw HTML that is contained in resend alert prompt for OV enrollment. 

## PR Checklist

- [x] Have you verified the basic functionality for this change?
- [x] Did you add tests, as appropriate, following our [Automated Test guidelines](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/2676497890/Automated+Testing+in+the+Signin+Widget)?
- [x] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?
- [ ] Did you verify the change by running [downstream monolith artifact](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/102897979/Sign-in+Widget+Development#Sign-inWidgetDevelopment-Instructionstocreateandrunthedownstreamartifact(d16t))? (Provide link to build below)
- [x] Does this PR include noticeable changes to the UI? (If yes, attach screenshots/video below)

### Issue:

- [OKTA-520236](https://oktainc.atlassian.net/browse/OKTA-520236)

### Screenshot/Video:

https://user-images.githubusercontent.com/97472729/184239564-860d9053-12a7-4f5e-aca4-9f931e4694ab.mov

Screen Shot 2022-08-11 at 4 30 08 PM

### CODE REVIEW

File: src/v3/src/transformer/oktaVerify/transformOktaVerifyEnrollPoll.ts [hunk 1]
```diff
@@ -89,16 +82,10 @@ export const transformOktaVerifyEnrollPoll: IdxStepTransformer = ({
       options: {
         // TODO: OKTA-520236 - i18n key has anchor tag
```
Comment (shawyu-okta) [inline src/v3/src/transformer/oktaVerify/transformOktaVerifyEnrollPoll.ts:83]: we can remove now :)

**Review by shawyu-okta — APPROVED** [review]

File: src/v3/src/types/schema.ts [hunk 2]
```diff
@@ -235,7 +234,7 @@ export interface ReminderElement extends UISchemaElement {
      */
     timeout?: number;
     linkLabel?: string;
-    action?: NextStep['action'];
+    className?: string;
```
Comment (glenfannin-okta) [inline src/v3/src/types/schema.ts:237]: Add additional property here to denote the TextWithHtml usage.

File: src/v3/src/components/ReminderPrompt/ReminderPrompt.tsx [hunk 3]
```diff
@@ -75,24 +80,38 @@ const ReminderPrompt: UISchemaElementComponent<{
...
     );
   };
 
-  const content = (
-    <Box marginBottom={2}>
-      {uischema.options?.ctaText}
-    </Box>
-  );
+  const renderAlertContent = () => {
+    if (className) {
```
Comment (glenfannin-okta) [inline src/v3/src/components/ReminderPrompt/ReminderPrompt.tsx:89]: Use newly added prop for condition of when to use TextWithHtml

File: src/v3/src/components/ReminderPrompt/ReminderPrompt.tsx [hunk 4]
```diff
@@ -15,22 +15,31 @@ import {
   Box,
   Link,
 } from '@mui/material';
-import { IdxActionParams } from '@okta/okta-auth-js';
 import { h } from 'preact';
 import { useEffect, useRef, useState } from 'preact/hooks';
 
-import { useWidgetContext } from '../../contexts';
-import { ReminderElement, UISchemaElementComponent, Undefinable } from '../../types';
+import { useOnSubmit } from '../../hooks';
```
Comment (glenfannin-okta) [inline src/v3/src/components/ReminderPrompt/ReminderPrompt.tsx:21]: Add integration test for this scenario.

File: src/v3/src/components/ReminderPrompt/ReminderPrompt.test.tsx [hunk 5]
```diff
@@ -66,28 +62,27 @@ describe('ReminderPrompt', () => {
...
-  it('should show prompt with working link after default timeout passes and include stateHandle when stateToken is set', async () => {
-    const mockStateHandle = 'abc12356789';
-    mockProps = { stateToken: '123abc' };
-    transaction.context = {
-      ...transaction.context,
-      stateHandle: mockStateHandle,
-    };
-    const mockActionFn = jest.fn();
-
+  it('should show prompt with working link after default timeout passes where ctaText contains HTML', async () => {
```
Comment (glenfannin-okta) [inline src/v3/src/components/ReminderPrompt/ReminderPrompt.test.tsx:75]: Add snapshot assertion as well (to each test)

File: src/v3/src/components/ReminderPrompt/ReminderPrompt.test.tsx [hunk 6]
```diff
@@ -66,28 +62,27 @@ describe('ReminderPrompt', () => {
...
+    const step = 'currentAuthenticator-resend';
     const props: UISchemaElementComponentProps & { uischema: ReminderElement; } = {
       uischema: {
         type: 'Reminder',
         options: {
-          ctaText: 'Didnt receive the email?',
-          linkLabel: 'Send again?',
-          action: mockActionFn,
+          ctaText: 'Didnt receive the email? Click <a href="#" class="send-again">send again</a>',
```
Comment (glenfannin-okta) [inline src/v3/src/components/ReminderPrompt/ReminderPrompt.test.tsx:81]: Add quote to `Didnt`

File: src/v3/src/types/schema.ts [hunk 7]
```diff
@@ -216,16 +215,16 @@ export interface DescriptionElement extends UISchemaElement {
...
     className: string;
-    step: string;
     stepToRender?: string;
   };
 }
 
 export interface ReminderElement extends UISchemaElement {
   type: 'Reminder';
-  options: {
+  options: ActionOptions & {
```
Comment (glenfannin-okta) [inline src/v3/src/types/schema.ts:227]: Change `ctaText` to `content` (and make new prop specific below, something like `isContentHtml` or `contentHasHtml`

Comment (glenfannin-okta) [reply src/v3/src/types/schema.ts:227]: update `linkLabel` to `buttonText`

**Pull request merged**